config/node_config.go

```diff
@@ -0,0 +1,451 @@
+package config
+
+import (
+	"encoding/json"
+
+	"code.vegaprotocol.io/vegacapsule/types"
+)
+
+/*
+description: |
+
+	Represents, and allows the user to configure, a set of Vega (with Tendermint) and Data Node nodes.
+	One node set definition can be used by applied to multiple node sets (see `count` field) and it uses
+	templating to distinguish between different nodes and names/ports and other collisions.
+
+example:
+
+	type: hcl
+	name: Node set with 2 validator nodes
+	value: |
+			node_set "validators" {
+				count = 2
+				mode = "validator"
+
+				node_wallet_pass = "n0d3w4ll3t-p4ssphr4e3"
+				vega_wallet_pass = "w4ll3t-p4ssphr4e3"
+				ethereum_wallet_pass = "ch41nw4ll3t-3th3r3um-p4ssphr4e3"
+
+				config_templates {
+					vega_file = "./path/vega_validator.tmpl"
+					tendermint_file = "./path/tendermint_validator.tmpl"
+				}
+			}
+*/
+type NodeConfig struct {
+	/*
+		description: |
+			Name of the node set.
+			Nomad instances that are part of these nodes are prefixed with this name.
+		example:
+			type: hcl
+			value: |
+				node_set "validators-1" {
+					...
+				}
+	*/
+	Name string `hcl:"name,label" cty:"name"`
+
+	/*
+		description: |
+			Determines what mode the node set should run in.
+		values:
+			- validator
+			- full
+	*/
+	Mode string `hcl:"mode" cty:"mode"`
+
+	/*
+		description: |
+			Defines how many node sets with this exact configuration should be created.
+	*/
+	Count int `hcl:"count" cty:"count"`
+
+	/*
+		description: Defines the password for the automatically generated node wallet associated with the created node.
+		required_if: mode=validator
+	*/
+	NodeWalletPass string `hcl:"node_wallet_pass,optional" template:"" cty:"node_wallet_pass"`
+
+	/*
+		description: Defines password for automatically generated Ethereum wallet in node wallet.
+		required_if: mode=validator
+	*/
+	EthereumWalletPass string `hcl:"ethereum_wallet_pass,optional" template:"" cty:"ethereum_wallet_pass"`
+
+	/*
+		description: Defines password for automatically generated Vega wallet in node wallet.
+		required_if: mode=validator
+	*/
+	VegaWalletPass string `hcl:"vega_wallet_pass,optional" template:"" cty:"vega_wallet_pass"`
+
+	/*
+		description: Whether or not Data Node should be deployed on node set.
+	*/
+	UseDataNode bool `hcl:"use_data_node,optional" cty:"use_data_node"`
+
+	/*
+		description: |
+					Path to [Visor](https://github.com/vegaprotocol/vega/tree/develop/visor) binary.
+					If defined, Visor is automatically used to deploy Vega and Data nodes.
```

Comment:
```suggestion
					If defined, Visor is automatically used to deploy Vega and Data nodes.
```
and update? (though I guess the updates are not automatic